### PR TITLE
change checkstyle & docs

### PR DESCRIPTION
**Pull request by CorvusYe**

### CODE REVIEW

**Review by wey-gu — APPROVED** [review]

**Review by wey-gu — COMMENTED** [review]
I see the rendered doc files(html) are also included, is this intended?

**Comment by CorvusYe** [comment]
> I see the rendered doc files(html) are also included, is this intended?

Yes, as a website container for md files.

**Comment by wey-gu** [comment]
> > I see the rendered doc files(html) are also included, is this intended?
> 
> Yes, as a website container for md files.

could we leave only the source of the doc(markdown) in this repo, let CI do the doc build, and put rendered HTML files in another branch(i.e. named gh-pages)?

**Comment by CorvusYe** [comment]
> could we leave only the source of the doc(markdown) in code master branch, let CI do the doc build, and put rendered HTML files in another branch(i.e. named gh-pages)?
> 
> There will be the version issue, though.

I have [another repo](https://github.com/CorvusYe/ngbatis-docs), and working on the version issue today.

image

**Comment by CorvusYe** [comment]
> could we leave only the source of the doc(markdown) in code master branch, let CI do the doc build, and put rendered HTML files in another branch(i.e. named gh-pages)?
> 
> There will be the version issue, though.

I have a question. How does GitHub turn a folder containing many MD files into a website. I haven't tried before.

**Comment by wey-gu** [comment]
> > could we leave only the source of the doc(markdown) in code master branch, let CI do the doc build, and put rendered HTML files in another branch(i.e. named gh-pages)?
> > There will be the version issue, though.
> 
> I have a question. How does GitHub turn a folder containing many MD files into a website. I haven't tried before.

From my understanding:

- GitHub pages just host static files
- the `turning MD files into a website` html files are just done by the (ssg static site generation) with [Jekyll](https://jekyllrb.com/), https://docusaurus.io/ , Hugo etc. we could use mike+mkdocs as https://github.com/vesoft-inc/nebula-docs-cn , and the generation would be done by CI(i.e. github actions)
  - while for JekyII, this could be done natively without extra github action workflow being added (doc: https://docs.github.com/cn/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll )

We can put HTML files now in master, in the future I could explore removing them and putting doc HTML files in another branch.

**Comment by CorvusYe** [comment]
> From my understanding:
> 
> * GitHub pages just host static files
> * the `turning MD files into a website` html files are just done by the (ssg static site generation) with [Jekyll](https://jekyllrb.com/), https://docusaurus.io/ , Hugo etc. we could use mike+mkdocs as https://github.com/vesoft-inc/nebula-docs-cn , and the generation would be done by CI(i.e. github actions)
>   
>   * while for JekyII, this could be done natively without extra github action workflow being added (doc: https://docs.github.com/cn/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll )
> 
> We can put HTML files now in master, in the future I could explore removing them and putting doc HTML files in another branch.

@wey-gu new skills, get ✅

**Review by wey-gu — APPROVED** [review]

**Comment by CorvusYe** [comment]
@wey-gu 
The commit of the document is reverted from this push. 
About document website, let's see in another pr.

**Comment by wey-gu** [comment]
> @wey-gu The commit of the document is reverted from this push. About document website, let's see in another pr.

Ok, thanks!

**Comment by wey-gu** [comment]
@CorvusYe in that PR, you could put only markdown things under docs, I will do the doc infra(making it into a doc site) thing afterwards.

I will take this chance to research and add dev docs for other projects, too.

**Comment by CorvusYe** [comment]
Any requirements about directory structure？If we want to implement i18n.

**Comment by wey-gu** [comment]
> Any requirements about directory structure？If we want to implement i18n.

You could put it as-is now, I will restructure it later when needed.

**Review by wey-gu — APPROVED** [review]
Thanks!!!

File: ngbatis-demo/src/test/java/ye/weicheng/ngbatis/demo/issues/TestIssue35.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import ye.weicheng.ngbatis.demo.repository.TestRepository;
+
+/**
+ * https://github.com/nebula-contrib/ngbatis/issues/35.
+ * @author yeweicheng
+ * @since 2022-09-01 20:20
+ * <br>Now is history!
+ */
+@SpringBootTest
+public class TestIssue35 {
```
Comment (Nicole00) [inline ngbatis-demo/src/test/java/ye/weicheng/ngbatis/demo/issues/TestIssue35.java:18]: Unit tests should be functional, and a specific problem can be used as a use case in the unit test.

please add license for a new file.

**Comment by Nicole00** [comment]
Another small suggestion, one PR try to only deal with one thing.

**Comment by CorvusYe** [comment]
@Nicole00 
Good idea.
I will pay more attention to the next pr which about "one PR one thing“.

**Review by Nicole00 — APPROVED** [review]
LGTM

**Pull request merged**